libr/bin/format/pe/pe.c

```diff
@@ -125,33 +142,110 @@ struct r_bin_pe_addr_t *PE_(check_msvcseh) (struct PE_(r_bin_pe_obj_t) *bin) {
 					const st32 call_dst = r_read_ble32 (b + n + 5, bin->big_endian);
 					entry->paddr += (n + 5 + 4 + call_dst);
 					entry->vaddr += (n + 5 + 4 + call_dst);
+					follow_jump (entry, bin->b, b, sizeof (b), bin->big_endian);
+					return entry;
+				}
+			}
+			//case5:
+			//57                                        push    edi
+			//56                                        push    esi
+			//FF 36                                     push    dword ptr[eax]
+			//E8 D9 FD FF FF                            call    _main
+			for (n = 0; n < sizeof (b) - 5; n++) {
+				if (b[n] == 0x57 && b[n + 1] == 0x56 && b[n + 2] == 0xff && b[n + 4] == 0xe8) {
+					const st32 call_dst = r_read_ble32 (b + n + 5, bin->big_endian);
+					entry->paddr += (n + 5 + 4 + call_dst);
+					entry->vaddr += (n + 5 + 4 + call_dst);
+					follow_jump (entry, bin->b, b, sizeof (b), bin->big_endian);
 					return entry;
 				}
 			}
+		}
+	}
 
+	// MSVC 32bit debug 
+	// push ebp
+	// mov ebp, esp
+	// call xxxxxxxx
+	if (b[3] == 0xe8) {
+		const st32 call_dst = r_read_ble32 (b + 4, bin->big_endian);
+		entry->paddr += (5 + 3 + call_dst);
+		entry->vaddr += (5 + 3 + call_dst);
+		r_buf_read_at (bin->b, entry->paddr, b, sizeof (b));
+		if (b[8] == 0xe8) {
+			const st32 call_dst = r_read_ble32 (b + 9, bin->big_endian);
+			entry->paddr += (5 + 8 + call_dst);
+			entry->vaddr += (5 + 8 + call_dst);
+			r_buf_read_at (bin->b, entry->paddr, b, sizeof (b));
+			if (b[0x152] == 0xe8) {
+				const st32 call_dst = r_read_ble32 (b + 0x153, bin->big_endian);
+				entry->paddr += (5 + 0x152 + call_dst);
+				entry->vaddr += (5 + 0x152 + call_dst);
+				r_buf_read_at (bin->b, entry->paddr, b, sizeof (b));
+				if (b[0x2e] == 0xe8) {
+					const st32 call_dst = r_read_ble32 (b + 0x2f, bin->big_endian);
+					entry->paddr += (5 + 0x2e + call_dst);
+					entry->vaddr += (5 + 0x2e + call_dst);
+					follow_jump (entry, bin->b, b, sizeof (b), bin->big_endian);
+					return entry;
+				}
+			}
 		}
 	}
+
 	// MSVC AMD64
 	// 48 83 EC 28       sub     rsp, 0x28
 	// E8 xx xx xx xx    call    xxxxxxxx
 	// 48 83 C4 28       add     rsp, 0x28
 	// E9 xx xx xx xx    jmp     xxxxxxxx
-	if (b[4] == 0xe8 && b[13] == 0xe9) {
-		//const st32 jmp_dst = b[14] | (b[15] << 8) | (b[16] << 16) | (b[17] << 24);
-		const st32 jmp_dst = r_read_ble32 (b + 14, bin->big_endian);
-		entry->paddr += (5 + 13 + jmp_dst);
-		entry->vaddr += (5 + 13 + jmp_dst);
-		if (r_buf_read_at (bin->b, entry->paddr, b, sizeof (b)) > 0) {
+	if (b[4] == 0xe8) {
+		bool found_caller = false;
+		if (b[13] == 0xe9) {
+			const st32 jmp_dst = r_read_ble32 (b + 14, bin->big_endian);
+			entry->paddr += (5 + 13 + jmp_dst);
+			entry->vaddr += (5 + 13 + jmp_dst);
+			found_caller = r_buf_read_at (bin->b, entry->paddr, b, sizeof (b)) > 0;
+		} else {
+			const st32 call_dst = r_read_ble32 (b + 5, bin->big_endian);
```

Comment:
The comment at https://github.com/radareorg/radare2/pull/17376/files#diff-15f63750bbf1cea750d78c8cfc33a916R196 seems to refer to the `if` case. What is the `else` matching exactly? Could you add a comment for this as well?